src/test/java/org/takes/rq/CapInputStreamTest.java

```diff
@@ -70,15 +69,35 @@ public void closesStream() throws Exception {
 
     /**
      * CapInputStream can skip on a stream.
-     * @throws Exception If some problem inside
+     * @throws IOException If some problem inside
      */
     @Test
-    public void skipsOnStream() throws Exception {
-        final long skip = 25L;
-        final InputStream stream = Mockito.mock(InputStream.class);
-        final CapInputStream wrapper = new CapInputStream(stream, 50L);
-        wrapper.skip(skip);
-        Mockito.verify(stream, Mockito.times(1)).skip(skip);
+    public void skipsOnStream() throws IOException {
+        final byte[] bytes = "skipsOnStream".getBytes();
+        final CapInputStream input = new CapInputStream(
+            new ByteArrayInputStream(bytes),
+            bytes.length
+        );
+        final long skipped = input.skip(2);
+        MatcherAssert.assertThat(skipped, Matchers.equalTo(2L));
+    }
+
+    /**
+     * CapInputStream decreases the available bytes after a skip call.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void skipDecreaseAvailableBytes() throws IOException {
+        final byte[] bytes = "skipDecreaseAvailableBytes".getBytes();
+        final CapInputStream input = new CapInputStream(
+            new ByteArrayInputStream(bytes),
+            bytes.length
+        );
+        final long skipped = input.skip(2);
+        MatcherAssert.assertThat(
```

Comment:
@rui-castro same remark as above, please inline everything, we have no need of `input` and `skipped` as they are used only once